SimplenoteWidgets/Models/Note+Widget.swift

```diff
@@ -30,3 +30,52 @@ extension Note {
     @NSManaged public var systemTags: String?
     @NSManaged public var tags: String?
 }
+
+extension Note {
+    private var lines: [String]? {
+        guard let content = content else {
+            return nil
+        }
+
+        return content.components(separatedBy: .newlines)
+    }
+
+    private var firstLine: String {
+        guard let lines = lines else {
+            return Constants.defaultTitle
+        }
+        return lines.first ?? String()
```

Comment:
You can simplify as follows:

```
lines?.first ?? Constants.defaultTitle
```

Reply:
This is serving two roles.  It is making sure that there is content in the note and that there is a title on the first line.  If we just check that there is a first line it could be \n which would make the note appear empty in the intents.  How about removing `private var first` and changing `var title` to:

```    var title: String {
        guard let firstLine = lines?.first,
              firstLine.count > 0 else {
            return Constants.defaultTitle
        }
        return firstLine
    }
```

That serves both roles in fewer lines/properties